e2etests/tests/unnumbered.go

```diff
@@ -0,0 +1,204 @@
+										ToAdvertise: frrk8sv1beta1.Advertise{
+											Allowed: frrk8sv1beta1.AllowedOutPrefixes{
+												Mode: frrk8sv1beta1.AllowAll,
+											},
+										},
+									},
+								},
+								Prefixes: []string{"5.5.5.5/32", "fc00:f888:ccd:e793::1/128"},
+							},
+						},
+					},
+					NodeSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"kubernetes.io/hostname": nodeWithP2PConnection.GetLabels()["kubernetes.io/hostname"],
+						},
+					},
+				},
+			}
+			err := updater.Update([]corev1.Secret{}, cr)
+			Expect(err).NotTo(HaveOccurred())
+			ginkgo.DeferCleanup(func() {
+				err := updater.DeleteFRRConfiguration(cr)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			validateUnnumberedBGPPeering(unnumberedExternalPeer, []string{"5.5.5.5", "fc00:f888:ccd:e793::1"})
+		})
+	})
+
+	// This spec is added to verify that the point-to-point setup works
+	ginkgo.Context("with raw config", func() {
+		ginkgo.It("session should be established and routes to be validated", func() {
+			raw := `
+		frr defaults traditional
+		no ipv6 forwarding
+		router bgp 65000
+			no bgp ebgp-requires-policy
+			no bgp network import-check
+			neighbor net0 interface remote-as external
+			address-family ipv6 unicast
+				network fc00:f888:ccd:e793::1/128
+				neighbor net0 activate
+			exit-address-family
+			address-family ipv4 unicast
+				network 5.5.5.5/32
+				neighbor net0 activate
+			exit-address-family
+		exit
+		end
+`
+
+			cr := frrk8sv1beta1.FRRConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "raw-for-" + unnumberedExternalPeer.Name,
+					Namespace: k8s.FRRK8sNamespace,
+				},
+				Spec: frrk8sv1beta1.FRRConfigurationSpec{
+					NodeSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"kubernetes.io/hostname": nodeWithP2PConnection.GetLabels()["kubernetes.io/hostname"],
+						},
+					},
+					Raw: frrk8sv1beta1.RawConfig{
+						Config:   raw,
+						Priority: 5,
+					},
+				},
+			}
+			ginkgo.DeferCleanup(func() {
+				err := updater.DeleteFRRConfiguration(cr)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			err := updater.Update([]corev1.Secret{}, cr)
+			Expect(err).NotTo(HaveOccurred())
+
+			validateUnnumberedBGPPeering(unnumberedExternalPeer, []string{"5.5.5.5", "fc00:f888:ccd:e793::1"})
+		})
+	})
+})
+
+func validateUnnumberedBGPPeering(peer *frrcontainer.FRR, prefixes []string) {
+	Eventually(func() error {
+		neighbors, err := frr.NeighborsInfo(peer)
+		if len(neighbors) == 0 {
+			return fmt.Errorf("must have at least one neighbor")
+		}
+		Expect(err).NotTo(HaveOccurred())
+		for _, n := range neighbors {
+			if !n.Connected {
+				return fmt.Errorf("node %v BGP session not established", n)
+			}
+		}
+		return nil
+	}, 2*time.Minute, 10*time.Second).ShouldNot(HaveOccurred(),
+		"timed out waiting to validate nodes peered with the frr instance")
+
+	// NOTE: we define the MAC address of net0, and therefore we can define the LLA
```

Comment:
where is this defined? can this be a const somewhere?

Reply:
I have refactor fully that part, now there is not hidden hardcoded connection (which looked ok when the function were in the same go file as the test and not part of a package)